controller.go

```diff
@@ -238,87 +209,52 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 
 	// Construct the network object
 	network := &network{
-		name:      name,
-		id:        types.UUID(stringid.GenerateRandomID()),
-		ctrlr:     c,
-		driver:    d,
-		endpoints: endpointTable{},
+		name:        name,
+		networkType: networkType,
+		id:          types.UUID(stringid.GenerateRandomID()),
+		ctrlr:       c,
+		endpoints:   endpointTable{},
 	}
 
 	network.processOptions(options...)
-	if err := c.addNetworkToStore(network); err != nil {
+
+	if err := c.addNetwork(network); err != nil {
 		return nil, err
 	}
-	// Create the network
-	if err := d.CreateNetwork(network.id, network.generic); err != nil {
+
+	if err := c.addNetworkToStore(network); err != nil {
```

Comment:
What should we do if `addNetworkToStore` fails? Should we roll back by letting the driver to delete network and removing the entry from local cache? Or just let the network be still available as a local entity?